### ISSUE TITLE
Coredump - 6.0.0 -   Expr::AsListExpr (const/<<>>)

### ISSUE THREAD

**Issue by initconf** [zeek/zeek#3263]
So I accidentally encountered this - here is (i) script to reproduce and (ii) backtrace.

```
module Test;
export {

        global a_table: table[subnet] of string = {
        1.2.3.4/24 = "unspecified",    # should have been [1.2.3.4/24] = "unspecified",
        };
}

```

```
Reading symbols from /usr/local/zeek/bin/zeek...
[New LWP 103082]
Core was generated by `zeek ./a.zeek'.
Program terminated with signal SIGABRT, Aborted.
Sent by thr_kill() from pid 30936 and user 20201.
#0  0x00000008037b1b8a in thr_kill () from /lib/libc.so.7
(gdb) bt
#0  0x00000008037b1b8a in thr_kill () from /lib/libc.so.7
#1  0x00000008037aff54 in raise () from /lib/libc.so.7
#2  0x0000000803728449 in abort () from /lib/libc.so.7
#3  0x0000000000ce9d21 in zeek::Reporter::FatalErrorWithCore (this=<optimized out>, fmt=<optimized out>) at /home/bro/install/zeek-6.0.0/src/Reporter.cc:154
#4  0x0000000000cf600f in zeek::Obj::BadTag (this=<optimized out>, msg=<optimized out>, t1=<optimized out>, t2=<optimized out>) at /home/bro/install/zeek-6.0.0/src/Obj.cc:112
#5  0x0000000000c7662d in zeek::detail::Expr::AsListExpr (this=0x807d07600) at /home/bro/install/zeek-6.0.0/src/Expr.cc:137
#6  zeek::detail::TableConstructorExpr::TableConstructorExpr (this=this@entry=0x807ceeab0, constructor_list=..., arg_attrs=..., arg_type=..., arg_attrs2=...) at /home/bro/install/zeek-6.0.0/src/Expr.cc:3606
#7  0x0000000000c8166f in zeek::make_intrusive<zeek::detail::TableConstructorExpr, zeek::IntrusivePtr<zeek::detail::ListExpr>&, decltype(nullptr), zeek::IntrusivePtr<zeek::Type>&>(zeek::IntrusivePtr<zeek::detail::ListExpr>&, decltype(nullptr)&&, zeek::IntrusivePtr<zeek::Type>&) (args=..., args=..., args=...)
    at /home/bro/install/zeek-6.0.0/src/include/zeek/IntrusivePtr.h:195
#8  0x0000000000c6a565 in zeek::detail::BinaryExpr::CheckForRHSList (this=this@entry=0x807cdc560) at /home/bro/install/zeek-6.0.0/src/Expr.cc:1332
#9  0x0000000000c7073b in zeek::detail::AssignExpr::AssignExpr (this=0x807cdc560, arg_op1=..., arg_op2=..., arg_is_init=<optimized out>, arg_val=..., attrs=..., typecheck=<optimized out>) at /home/bro/install/zeek-6.0.0/src/Expr.cc:2524
#10 0x0000000000d7634d in zeek::make_intrusive<zeek::detail::AssignExpr, zeek::IntrusivePtr<zeek::detail::NameExpr>&, zeek::IntrusivePtr<zeek::detail::Expr>&, bool> (args=@0x7fffffffa397: false, args=@0x7fffffffa397: false, args=@0x7fffffffa397: false) at /home/bro/install/zeek-6.0.0/src/include/zeek/IntrusivePtr.h:195
#11 0x0000000000d71633 in zeek::detail::initialize_var (id=..., c=zeek::detail::INIT_FULL, init=...) at /home/bro/install/zeek-6.0.0/src/Var.cc:176
#12 zeek::detail::make_var (id=..., t=..., c=<optimized out>, c@entry=zeek::detail::INIT_FULL, init=..., attr=std::unique_ptr<std::__1::vector<zeek::IntrusivePtr<zeek::detail::Attr>, std::__1::allocator<zeek::IntrusivePtr<zeek::detail::Attr> > >> = 0x0, dt=dt@entry=zeek::detail::VAR_REGULAR, do_init=<optimized out>)
    at /home/bro/install/zeek-6.0.0/src/Var.cc:358
#13 0x0000000000d70a5d in zeek::detail::add_global (id=..., t=..., c=zeek::detail::INIT_NONE, c@entry=zeek::detail::INIT_FULL, init=..., attr=std::unique_ptr<std::__1::vector<zeek::IntrusivePtr<zeek::detail::Attr>, std::__1::allocator<zeek::IntrusivePtr<zeek::detail::Attr> > >> = 0x0, dt=dt@entry=zeek::detail::VAR_REGULAR)
    at /home/bro/install/zeek-6.0.0/src/Var.cc:408
#14 0x0000000000adfc2b in build_global (id=0x807cf5280, t=0x807cee970, ic=zeek::detail::INIT_FULL, e=0x807cee9c0, attrs=<optimized out>, dt=dt@entry=zeek::detail::VAR_REGULAR) at parse.y:322
#15 0x0000000000ad9145 in yyparse () at parse.y:1368
#16 0x0000000000bfe719 in zeek::detail::setup (argc=argc@entry=2, argv=argv@entry=0x7fffffffeb70, zopts=0x2, zopts@entry=0x0) at /home/bro/install/zeek-6.0.0/src/zeek-setup.cc:808
#17 0x000000000074f857 in main (argc=6, argv=0x0) at /home/bro/install/zeek-6.0.0/src/main.cc:57
```

Let me know if any questions - I can provide more info/background.

**Comment by bbannier** [comment]
Looks like this got broken by #1999 which landed first in 5.0.0.

**Comment by ckreibich** [comment]
I can confirm this — I didn't bisect, but with our 4.2.2 Docker image one still gets the expected:
```
error in ././test.zeek, line 5: illegal initializer (1.2.3.0/24 = unspecified)
```
